controllers/certificatesigningrequests_controller.go

```diff
@@ -19,45 +19,195 @@ package controllers
+			// Return and don't requeue
+			return ctrl.Result{}, nil
+		}
+		// Error reading the object - requeue the request.
+		return ctrl.Result{}, err
+	}
+
+	return ctrl.Result{}, r.reconcileCSR(certificateSigningRequest, configMap)
+}
 
-	// your logic here
+// NewCertificateSigningRequestsReconciler returns a pointer to certificateSigningRequestsReconciler
+func NewCertificateSigningRequestsReconciler(mgr manager.Manager, clusterConfig cluster.Config, watchNamespace string) (*certificateSigningRequestsReconciler, error) {
+	clientset, err := kubernetes.NewForConfig(mgr.GetConfig())
+	if err != nil {
+		return nil, errors.Wrap(err, "error creating kubernetes clientset")
+	}
 
-	return ctrl.Result{}, nil
+	return &certificateSigningRequestsReconciler{
+			instanceReconciler: instanceReconciler{
+				client:             mgr.GetClient(),
+				log:                ctrl.Log.WithName("controllers").WithName("CertificateSigningRequests"),
+				k8sclientset:       clientset,
+				clusterServiceCIDR: clusterConfig.Network().GetServiceCIDR(),
+				vxlanPort:          "",
+				watchNamespace:     watchNamespace,
+			},
+			platform: clusterConfig.Platform(),
+		},
+		nil
+}
+
+// reconcileCSR handles the CSR validation and approval
+func (r *certificateSigningRequestsReconciler) reconcileCSR(request *certificatesv1.CertificateSigningRequest, configMap *core.ConfigMap) error {
+	if request == nil || configMap == nil {
+		return errors.New("CSR or instance ConfigMap cannot be nil")
+	}
+
+	hostNames, err := r.findHostNames(configMap)
+	if err != nil {
+		return errors.Wrapf(err, "unable to find host names from ConfigMap %s", configMap.Name)
+	}
+
+	certificateSigningRequest, err := csr.NewApprover(request, r.k8sclientset, hostNames, r.log)
+	if err != nil {
+		return errors.Wrapf(err, "could not create WMCO CSR Approver")
+	}
+
+	if err := certificateSigningRequest.Approve(); err != nil {
+		return errors.Wrapf(err, "WMCO CSR Approver could not approve %s CSR", request.Name)
+	}
+
+	return nil
+}
+
+// findHostNames returns a list of the instance host names from the instance configMap
+// depending on the cloud provider type of the cluster.
+func (r *certificateSigningRequestsReconciler) findHostNames(configMap *core.ConfigMap) ([]string, error) {
+	// Get the list of hosts that are expected to be Nodes
+	hosts, err := parseHosts(configMap.Data)
+	if err != nil {
+		return nil, errors.Wrapf(err, "unable to parse hosts from ConfigMap %s", configMap.Name)
+	}
+	var requireVMHostName bool
+	// platform type set None or vSphere indicates hostName required is the actual hostName of the VM
+	if r.platform == oconfig.NonePlatformType || r.platform == oconfig.VSpherePlatformType {
+		requireVMHostName = true
+	}
+
+	var hostNames []string
+	for _, host := range hosts {
+		hostName := host.Address
+		if requireVMHostName {
+			nc, err := nodeconfig.NewNodeConfig(r.k8sclientset, r.clusterServiceCIDR, r.vxlanPort, host, r.signer,
```

Comment:
yes that's what I am using nodeconfig for but I used it here to avoid getting into populating `nodeConfigCache.workerIgnitionEndPoint ` when calling https://github.com/openshift/windows-machine-config-operator/blob/4a90df34920a4c295c9cfefaa896cfa799576748/pkg/nodeconfig/nodeconfig.go#L114

Reply:
Elaborated more on my thoughts offline, quick recap:
* Nodeconfig has a purpose of Configuring and Deconfiguring nodes, it having a `Windows` interface present in it is a means to achieving that.
* In this case what is needed is not configuring instances but being able to run a single command against an instance, for that, instantiating a Windows object is more proper.
* We shouldnt need `workerIgnitionEndPoint` to run a command against an instance, but that is a current restriction put in place by the implementation of `windows.New`
* If removing that restriction is too much work, I am fine with the change not being made, but I think a TODO and an associated ticket should be created for this.